### DISCUSSION TITLE
how to automatically scroll pygubu scrolledframe to end of input text?

### DISCUSSION THREAD

**Discussion by AJ6CU** [alejandroautalan/pygubu-designer#185] [Q&A] [answered]
Using an instance of the pygubu scrolledframe widget to contain a tk.Text. The Tet object is basically a log that I write to. Is there a way to autoscroll the window so that the latest output is always shown? Perhaps I have the height missmatched between the scrolledFrame and the Text widget? I assume I could have a much larger text widget and that the scrollframe would provide a window into that text widget. Only need the vertical (y) to keep up with the ongoing log.

Thanks for any help!

Mark

**Comment by alejandroautalan** [comment, marked as answer]
Hello  Mark,
> The Tet object is basically a log that I write to. Is there a way to autoscroll the window so that the latest output is always shown? 

Just use the method "see" of the tk.Text widget after adding your text log.

    txt.see("end")

> Using an instance of the pygubu scrolledframe widget to contain a tk.Text

If you are using scrolledframe only to add scrollbars to tk.Text is better to use the ScrollbarHelper.

I created and [example](https://github.com/alejandroautalan/pygubu-designer/tree/master/examples/text/logwindowdemo) using a tk.Text with a ScrollbarHelper.

Regards
Alejandro A.

**Comment by AJ6CU** [comment]
Alejandro,

Your example worked perfectly!  Thank you.  Here is a screen capture of the tool I am working on for the uBITX ham radio firmware. More and more of this is being generated just by your tool. Thank you!

Mark
AJ6CU
![ubitx_customizer_v2 0screencapture](https://user-images.githubusercontent.com/70183884/210596727-59a3af77-bd3d-4dfc-a3ab-7201f1754c0f.JPG)

**Reply by alejandroautalan** [reply]
I'm glad pygubu is useful for you. Regards!

**Reply by jrezai** [reply]
@AJ6CU Nice window!
@alejandroautalan Thanks for Pygubu and happy new year!

**Reply by alejandroautalan** [reply]
Thank you @jrezai, happy new year to you too!

**Comment by AJ6CU** [comment]
Happy New Years to you all too!  Thank you for making this wonderful tool available. MarkSent from my iPhoneOn Jan 4, 2023, at 2:03 PM, Alejandro Autalán ***@***.***> wrote:﻿
Thank you @jrezai, happy new year to you too!
